### ISSUE TITLE
Request: Custom-building a new record in controller

### ISSUE THREAD

**Issue by mcmire** [thoughtbot/administrate#220]
I have a form that contains nested attributes for an association. In order to render the `new` page, I need a way to pre-populate the association on the record with an empty object. Right now, I can override the `new` method like so:

``` ruby
def new
  new_resource = resource_class.new
  new_resource.my_has_many_association.build
  @page = Administrate::Page::Form.new(dashboard, new_resource)
end
```

However, I'm wondering if it would be better if there were a `new_resource` method I could simply override:

``` ruby
def new_resource
  resource_class.new.tap do |resource|
    resource.my_has_many_association.build
  end
end
```

Thoughts?

**Comment by graysonwright** [comment]
@mcmire, that's a good idea. Also note that your solution won't work on the current version of Administrate, because we've moved away from instance variables in the controller.

**Comment by mcmire** [comment]
Whoops! You're right. Well, you get the picture.

**Comment by GustavoCaso** [comment]
@graysonwright is anyone working on this one ? if not I will like to tackle this one

**Comment by carlosramireziii** [comment]
@GustavoCaso Not sure if you're still interested in tackling this one, but if you are let me know.

**Comment by GustavoCaso** [comment]
@tysongach sorry for the really late response 😄  I will check for a solution on this issue

**Comment by nickcharlton** [comment]
@GustavoCaso Were you able to look into this?

**Comment by GustavoCaso** [comment]
I have. But I haven’t finish with it. Currently I’m not at home, since I’m on holidays. If it is urgent I will be able to push something in two weeks :)

**Comment by nickcharlton** [comment]
Hi @GustavoCaso! No worries. Take your time, I look forward to seeing it!

**Comment by nickcharlton** [comment]
I'm going to close this as we merged #1097 the other month and so this should be done now!
